### PR TITLE
Add possibility to stop Netty 4 Timer

### PR DESCRIPTION
**Pull request by DanielCharczynski**
Problem

No possibility to stop "Netty 4 Timer".
Detailed description is available here: https://github.com/twitter/finagle/issues/757

Solution

Make HashedWheelTimer object public with additional method to stop "Netty 4 Timer"

Result

Possibility to stop "Netty 4 Timer"

### CODE REVIEW

File: finagle-netty4/src/main/scala/com/twitter/finagle/netty4/package.scala [hunk 1]
```diff
@@ -15,4 +15,10 @@ package object netty4 {
    */
   private[finagle] val Toggles: ToggleMap =
     StandardToggleMap("com.twitter.finagle.netty4", DefaultStatsReceiver)
+	
+  /** Stops HashedWheelTimer thread */
+  def stopHashedWheelTimer {
```
Comment (mosesn) [inline finagle-netty4/src/main/scala/com/twitter/finagle/netty4/package.scala]: how about we instead make the HashedWheelTimer object public and add a public method `stop` to it?  I think that will be a cleaner API.

File: finagle-netty4/src/main/scala/com/twitter/finagle/netty4/Netty4HashedWheelTimer.scala [hunk 2]
```diff
@@ -24,7 +24,7 @@ private object timerTickDuration
 /**
  * A Netty timer for use with [[Netty4HashedWheelTimer]].
  */
-private class HashedWheelTimer(
+class HashedWheelTimer(
```
Comment (mosesn) [inline finagle-netty4/src/main/scala/com/twitter/finagle/netty4/Netty4HashedWheelTimer.scala]: can we do it while keeping this private?  or does that cause a problem?  if we can't make this private, can we make the constructor private?

File: finagle-netty4/src/main/scala/com/twitter/finagle/netty4/Netty4HashedWheelTimer.scala [hunk 3]
```diff
@@ -67,7 +67,7 @@ private class HashedWheelTimer(
   self.newTimeout(pendingTasksStat, statsPollInterval.inSeconds, TimeUnit.SECONDS)
 }
 
-private object HashedWheelTimer {
+object HashedWheelTimer {
```
Comment (mosesn) [inline finagle-netty4/src/main/scala/com/twitter/finagle/netty4/Netty4HashedWheelTimer.scala]: thinking about it more, I think that we should probably make `instance` private, and just have a stop method on this object.  that way people can't pass around HashedWheelTimer willy-nilly and accidentally get confused and stop it.  you have to know that you're stopping the default one.

Comment (DanielCharczynski) [reply finagle-netty4/src/main/scala/com/twitter/finagle/netty4/Netty4HashedWheelTimer.scala]: Yes, that was one of my ideas. Done

**Comment by mosesn** [comment]
Looks great, thanks!

**Comment by jyanJing** [comment]
Thank you for working on this @DanielCharczynski ! The change is merged in [fa4899e](https://github.com/twitter/finagle/commit/fa4899e1ba1e9b612dbf7428b0f457ae3855b5b8)

**Pull request closed without merging**